### PR TITLE
crimson/os/seastore: track retired extents until in-progress transactions complete

### PR DESCRIPTION
**Pull request by athanatos**
<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by athanatos** [comment]
@liu-chunmei I think this may fix the crash you were seeing with crimson-store-nbd.

**Comment by cyx1231st** [comment]
The issue I've seen is still there after rebasing onto this PR, which seems related to the retire path in the lba tree.

To reproduce: https://gist.github.com/cyx1231st/fae92aa0b3825d9e45891f73fadeb78c   @athanatos Do you have clue what might be going wrong?

I think the onode-tree-side implementation is working because this test is passing under "Dummy" backend by changing to `TEST_SEASTORE` to `false` in https://github.com/cyx1231st/ceph/blob/99c226f7143afaad41280c0f73cac79283bc6fc1/src/test/crimson/seastore/onode_tree/test_staged_fltree.cc#L1520

**Comment by liu-chunmei** [comment]
> @liu-chunmei I think this may fix the crash you were seeing with crimson-store-nbd.

I will verify it soon.

**Comment by athanatos** [comment]
Same crash seems to happen without this PR, seems to be a bug in address resolution in the lba btree.  Investigating.

**Comment by athanatos** [comment]
@cyx1231st Try the two commits at the top of https://github.com/athanatos/ceph/tree/sjust/wip-fix-addr when you get a chance.

**Comment by cyx1231st** [comment]
It works great. I see no issues now.

**Review by tchaikov — APPROVED** [review]

File: src/crimson/os/seastore/seastore_types.h [hunk 1]
```diff
@@ -60,6 +62,8 @@ std::ostream &segment_to_stream(std::ostream &, const segment_id_t &t);
 using segment_off_t = int32_t;
 constexpr segment_off_t NULL_SEG_OFF =
   std::numeric_limits<segment_id_t>::max();
+constexpr segment_off_t MAX_SEG_OFF =
+  std::numeric_limits<segment_id_t>::max();
```
Comment (liu-chunmei) [inline src/crimson/os/seastore/seastore_types.h:66]: does it std::numeric_limits<segment_off_t>::max();

**Review by liu-chunmei — APPROVED** [review]
only need a small fix see comment.  nbd still hang when iodepth=32 run time set 10mins.  I am still thinking how to locate where hang.

**Pull request merged**